### PR TITLE
Spreedly: Support gateway_specific_response_fields within the reponse parameters

### PR DESCRIPTION
**Pull request by abarrak**

### CODE REVIEW

**Comment by abarrak** [comment]
@bpollack what's the status on the PR?

Parsing `gateway_specific_response_fields` then return it with response reflects the full API return.

**Comment by abarrak** [comment]
This is being open for a while now and used in our fork, @curiousepic @jknipp would you take a look for merging?
Thanks.

File: lib/active_merchant/billing/gateways/spreedly_core.rb [hunk 1]
```diff
@@ -244,11 +244,20 @@ def parse(xml)
...
-          response[name.to_sym] = Hash.from_xml(childnode.to_s).values.first
+        node_name = node.name.downcase
+        childnode_name = childnode.name.downcase
+        composed_name = "#{node_name}_#{childnode_name}"
+
+        childnodes_present = !childnode.elements.empty?
+
+        if childnodes_present && composed_name == 'payment_method_data'
+          response[composed_name.to_sym] = Hash.from_xml(childnode.to_s).values.first
+        elsif childnodes_present && node_name == 'gateway_specific_response_fields'
```
Comment (abarrak) [inline lib/active_merchant/billing/gateways/spreedly_core.rb:255]: Hi @curiousepic would you take a look here?
This is to enabled additional `gateway_specific_response_fields` which isn't returning by default in Active Merchant unlike the [direct Spree Gateway integration](https://docs.spreedly.com/payment-gateways/migs/#gateway-specific-response-fields).

**Comment by curiousepic** [comment]
@abarrak  Apologies on the silence here - if you rebase from latest master, ensure tests still pass, and reopen it I will try to get it in our pipeline for review and merging!

**Comment by abarrak** [comment]
Hi @curiousepic,
Rebased the commit with master branch and tests in [PR here](https://github.com/activemerchant/active_merchant/pull/4064).

**Pull request closed without merging**